### ISSUE TITLE
Output Locations as a function of the display conditions

### ISSUE THREAD

**Issue by willcitizen** [pojome/elementor#5324]
## Prerequisites

- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [x] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**

This is a small irritation, but an irritation none-the-less.  

When you start to register additional locations and are protoyping designs the need might arise to display a template in a different location from where you created it (for example a 'below-header' item might move into an location named 'banner', or something similar) Currently to achieve this you have to create a new template in the new location, copy and paste the content in and reassign display conditions.  

**Describe the solution you'd like**
Instead of working with a fixed template type, bound to A registered location, it would make more sense to wrap the location into the display conditions logic.  (display this template at location x on pages y and z)

This would also allow for more fluid design choices and reduce content duplication and the need for excessive template nesting - consider the example: 

Display this banner in location 'above-content' on pages x y and z, but display it in location 'below-content on pages e,d and f)

In order to achieve that with the current solution I would have to create my banner in a separate template and then create two additional templates, one for above-content and one for below-content and the only job of those templates would be to act as a wrapper for the banner and control its output conditions and page position.

**Describe alternatives you've considered**
There are no alternatives other than the inefficient nesting scenarios I have described that would achieve this.

**Additional context**

**Comment by bainternet** [comment]
@willcitizen 

I understand what you are suggesting but its simply not the way locations work.

A different approach which we are strongly thinking about (like we discussed on a different issue) is to allow visibility conditions/rule per widget and this will allow you to include the template via a widget/shortcode and set your conditions.

thanks for the suggestion.

**Comment by willcitizen** [comment]
Thanks @bainternet  that makes sense to me - it's a more elegant solution in the long run :) 

I did find the Locations Settings Option in the editor which actually does allow you to move a template between locations since I posted this (I suspect that only becomes visible once you have custom locations configured so I didn't spot it before)

I really hope we will see something like this and some documentation for the conditions manager in the near future - I have been checking out the code for it to see if I could work out how to extend it myself, but my oop php is a little rusty.

![image](https://user-images.githubusercontent.com/41669565/43839905-9318416e-9b1f-11e8-8fd2-8809d30cbf67.png)

While I'm waiting to see which way you guys take this I hacked together my own location manager using an ACF options page, a couple of functions to handle registration and a location output shortcode.  I'm finding this works really well as I can output my custom locations inside the core header/single/footer and use a few taxonomy toggles on my pages to flip between different page-layouts, or toggle areas on and off.
